src/main/java/net/minecraftforge/registries/ObjectHolderRegistry.java

```diff
@@ -150,11 +150,11 @@ private void scanClassForFields(Map<String, String> classModIds, String classNam
             {
                 continue;
             }
-            addHolderReference(new ObjectHolderRef(f, new ResourceLocation(value, f.getName()), extractFromExistingValues));
+            addHolderReference(new FieldObjectHolderRef(null, f, new ResourceLocation(value, f.getName()), extractFromExistingValues));
         }
     }
 
-    private void addHolderReference(ObjectHolderRef ref)
+    public void addHolderReference(IObjectHolderRef ref)
     {
```

Comment:
If we're exposing this publicly, we need some form of deduplication. Don't want people calling this 10x for the same fields...

Reply:
I think a generic `ClassMember` sort of value type would be good for this. That is to say a simple object consisting of a class (name) reference, a member name and some type information. The method can then simply check against a map/set of these member references.

Reply:
Any update on implementing the deduplication?